File: RelationExample/RelationExample/Square.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RelationExample
{
    class Square : Quadrangle
    {

        public Square(float sideLengthA) : base(sideLengthA, sideLengthA, sideLengthA, sideLengthA)
        {}

        public float CalculateArea()
        {
            return sideLengthA * sideLengthA;
        }

        public override void PrintPerimeter(float perimeter)
        {
            Console.WriteLine("The perimeter of the square is: " + perimeter);
        }

        public override void PrintArea(float area)
        {
            Console.WriteLine("The area of the square is: " + area);
        }
    }
}

File: RelationExampleC++/RelationExampleC++/Square.cpp
#include "Square.h"

Square::Square(float sideLengthA) : Quadrangle(sideLengthA, sideLengthA, sideLengthA, sideLengthA)
{}

float Square::CalculateArea()
{
	return m_sideLengthA * m_sideLengthA;
}

void Square::PrintPerimeter(float perimeter)
{
	std::cout << "The perimeter of the square is: " << perimeter << "\n";
}

void Square::PrintArea(float area)
{
	std::cout << "The area of the square is: " << area << "\n";
}